### ISSUE TITLE
Move “git extras” (i.e. git blame etc.) extension to core workflow

### ISSUE THREAD

**Issue by ryankscott** [sourcegraph/sourcegraph#38377]
- [x] https://github.com/sourcegraph/sourcegraph/issues/39433
- [x] https://github.com/sourcegraph/sourcegraph/issues/39437
- [x] https://github.com/sourcegraph/sourcegraph/issues/41314

**Comment by sourcegraph-bot-2** [comment]
Heads up @muratsu @jjinnii @ryankscott - the "team/integrations" label was applied to this issue.

**Comment by taras-yemets** [comment]
There are three pages where we show git blame decorations: file view, commit view (split and unified), and compare view.
The file view already has column implementation for the blame decorations (as an experimental feature - https://github.com/sourcegraph/sourcegraph/pull/36007). The other two (commit and compare views) currently support only inline decorations - we were going to revisit it after we collect some feedback on the blame column view. 

There are a few design questions I'd like to clarify before moving forward with the implementation.
1. Git-extras, open in editor, and search export are the only extensions we're going to migrate ([FAQ](https://docs.google.com/document/d/1wZqz7C1u6bnxYyEktILt4jJyLLkSBa8DIJLO_cvxpUM/edit#bookmark=id.2we9ig21ysxi)). What do you think, should we move git-extras and open in editor icons next to the code host icon (search export button is already in that row) and remove the existing action items bar?
2. Are we going to change the blame decorations display on commit and compare views?

<details>
<summary>Screenshots</summary>
<table>
<tbody>
<tr>
<td>File view</td>
<td>
<img width="2544" alt="Screenshot 2022-07-22 at 11 06 53" src="https://user-images.githubusercontent.com/25318659/180401386-ec810164-6811-4d73-9ac6-bff6d4d9ba08.png">

</td>
</tr>
<tr>
<td>Commit view</td>
<td>
<img width="2542" alt="Screenshot 2022-07-22 at 11 09 07" src="https://user-images.githubusercontent.com/25318659/180401816-a1473a2e-f678-4e68-b9f7-ff120bc24c64.png">

<img width="2541" alt="Screenshot 2022-07-22 at 11 07 57" src="https://user-images.githubusercontent.com/25318659/180401528-4741d0da-dcd5-4192-bc16-412927754987.png">

</td>
</tr>
<tr>
<td>Compare view</td>
<td>
<img width="2545" alt="Screenshot 2022-07-22 at 11 15 53" src="https://user-images.githubusercontent.com/25318659/180401506-9d5e41ac-3cbe-44a1-92b1-58ae42e0cfb4.png">
</td>

</tr>
</tbody>
</table>
</details>

cc: @jjinnii

**Comment by jjinnii** [comment]
Hey @taras-yemets ! Replies to your questions: 

1. I am currently working through designs on where we might incorporate all selected extensions into our existing page structure. This is something that requires [systematic design](https://app.excalidraw.com/s/4Dr1S6qmmY7/1PUMkKyEc1Z), rather than adding in on an extensions by extension basis. In addition, any changes to where/how a user can take action on the file view page is something that requires close collaboration with other product teams. 
TL;DR feel free to move forward with any placeholder location as I work through designs on this, if this is blocking your work, but keep in mind that it will likely require some additional work later on.

2. Similar as above, I don't have an answer yet as I haven't worked through the individual git blame designs. I think likely yes, as we need to have a higher standard for consistency within our core product. Again, this can be something we revisit after I've had a chance to think through the problem holistically. (Should be within this week). 

My understanding is that engineering is working ahead on migrating functionality into the core product, but design is necessarily behind as this happens. I will be working on blame/extension designs this week so I'll have answers for you hopefully soon. 😄

**Comment by jjinnii** [comment]
@taras-yemets Documenting design guidance on blame for 4.0: 

- Remove blame support on commit/compare views. Reasoning: GH/GL does not support (not an industry expected standard), and[ internal feedback](https://sourcegraph.slack.com/archives/C03CSAER9LK/p1659686650896819) that it's not discoverable/not expected
- [Updated designs](https://www.figma.com/file/7bVINrmqfDpxmLx3gv1uVa/Sunsetting-extensions?node-id=2116%3A22361) that include a new blame-specific popover. Please also note the changes to font.

As discussed in Figma, I've also included stretch/post 4.0 designs for the blame popover that pulls in new PR/MR functionality, as well as a slightly wider column width. 

cc: @ryankscott

**Comment by philipp-spiess** [comment]
@taras-yemets @jjinnii  Given that we have an issue for the remaining task (https://github.com/sourcegraph/sourcegraph/issues/39437), do you think we can close this one?

**Comment by jjinnii** [comment]
@philipp-spiess Does #39437 also encompass the new design changes that are documented in this issue as well? It seems as if that issue only covers removing blame as an experimental feature?

**Comment by philipp-spiess** [comment]
@jjinnii Oh, sorry, my bad. You're right this still needs design changes.
